platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +181,25 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                IOUtils.copy(is, baos);
+                byte[] bytes = baos.toByteArray();
+                is = new ByteArrayInputStream(bytes);
+
+                Parser parser = new AutoDetectParser();
+                ToXMLContentHandler xmlContentHandler = new ToXMLContentHandler();
+                Metadata metadata = new Metadata();
+                parser.parse(is, xmlContentHandler, metadata, new ParseContext());
+                is.reset();
```

Comment:
When you switch this over to use the file backed stream, you can simply ask it for a new stream as opposed to resetting the existing one.